praw/models/reddit/message.py

```diff
@@ -95,6 +108,22 @@ def delete(self):
             API_PATH["delete_message"], data={"id": self.fullname}
         )
 
+    def __setattr__(self, name: str, value: Union[str, Dict[str, Any]]):
+        """Objectify author, subreddit and dest."""
+        if name == "author" and isinstance(value, dict):
+            value = Redditor(self._reddit, _data=value)
+        if name == "subreddit" and isinstance(value, dict):
+            value = Subreddit(self._reddit, _data=value)
+        if name == "dest" and isinstance(value, dict):
+            # Things get tricky. Dest can be a redditor or a subreddit.
+            # However, subreddits have key ``display_name`` and redditors have
```

Comment:
Why not just objectify `value` and then check the return value's type?

Reply:
I have no clue how to use the Reddit objectifier, as there isn’t too much code on it and the objectifier code is very non-obvious.

Reply:
It's pretty simple to use — just give it a `dict` that represents some object, and it will give you an object in return. Though the code of the method isn't great, I do think it's a positive that all logic regarding deciding what any given chunk of data represents is contained in one place. This avoids having to duplicate this sort of decision, as you do here. That's a good thing because when we need to change the way a certain model is objectified, we know that we only have to make the change in one place.

For an example, see: https://github.com/praw-dev/praw/blob/76dbc43e722d355baff4d30ae06cd7333f4ddd91/praw/models/reddit/modmail.py#L168